event_sale_seat_reserve/models/sale_order_line.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0).
+
+from odoo import models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _update_registrations(
+        self,
+        confirm=True,
+        cancel_to_draft=False,
+        registration_data=None,
+        mark_as_paid=False,
+    ):
+        res = super(SaleOrderLine, self)._update_registrations(
+            confirm=confirm,
+            cancel_to_draft=cancel_to_draft,
+            registration_data=registration_data,
+            mark_as_paid=mark_as_paid,
+        )
+        RegistrationSudo = self.env["event.registration"].sudo()
+        registrations = RegistrationSudo.search(
+            [("sale_order_line_id", "in", self.ids), ("state", "=", "draft")]
+        )
+        for registration in registrations:
+            if registration._need_pre_reservation():
```

Comment:
`def _need_pre_reservation(self):` always returns False in "event.registration" model (file event_seat_reserve/models/event_registration.py in this PR) or am I missing something ? :thinking: